src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,166 @@
+import { MouseEventHandler, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import ReactDatePicker from 'react-datepicker';
+
+import { Svg } from 'components/Svg';
+import { variables } from 'styles/variables';
+import { StyledBodyLarge } from 'styles/styledComponents/Typography';
+
+import {
+  StyledButton,
+  StyledButtons,
+  StyledCollapseBtn,
+  StyledIconBtn,
+  StyledPopover,
+  StyledTextField,
+} from './DatePicker.styles';
+import { DateVariant, DatePickerProps, UiType, MinMaxDate } from './DatePicker.types';
+import { DatePickerHeader } from './DatePickerHeader';
+import { getDatesStringsArray, getStringFromDate, getDateFromString } from './DatePicker.utils';
+import { datePlaceholder } from './DatePicker.const';
+
+export const DatePicker = ({ value, setValue, uiType = UiType.oneDate }: DatePickerProps) => {
+  const { t } = useTranslation('app');
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [showFirstCalendar, setShowFirstCalendar] = useState(true);
+  const [showSecondCalendar, setShowSecondCalendar] = useState(false);
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'date-picker-popover' : undefined;
+  const handlePickerShow: MouseEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    setAnchorEl(event.currentTarget);
+  };
+  const handlePickerClose = () => setAnchorEl(null);
+  const handleCancelClick = () => {
+    setValue('');
+    handlePickerClose();
+  };
+
+  const handleShowFirstClick = () => {
+    setShowFirstCalendar((prevState) => !prevState);
+    setShowSecondCalendar(false);
+  };
+
+  const handleShowSecondClick = () => {
+    setShowFirstCalendar(false);
+    setShowSecondCalendar((prevState) => !prevState);
+  };
+
+  const handleDateChange = (date: Date | null, variant?: DateVariant) => {
+    if (uiType === UiType.oneDate) {
+      setValue(date ? getStringFromDate(date) : '');
+    } else {
+      const valueArr = getDatesStringsArray(value);
+      if (date) {
+        const startDate = variant === 'start' ? getStringFromDate(date) : valueArr?.[0];
+        const endDate = variant === 'end' ? getStringFromDate(date) : valueArr?.[1];
+
+        setValue(`${startDate || datePlaceholder} - ${endDate || datePlaceholder}`);
+      }
+    }
+  };
+
+  const getSelectedDate = (variant?: DateVariant) => {
+    if (uiType === UiType.oneDate) {
+      return value ? getDateFromString(value) : new Date();
+    } else {
+      let resultVal = new Date();
```

Comment:
maybe it's better to rename resultVal to "selectedDate" and valueArr to "dateArray"

Reply:
done